### PR TITLE
Add docs, adjust types and change android linking

### PR DESCRIPTION
**Pull request by Esemesek**
# Summary

- [x] - Added documentation
- [x] - Fixed Android linking
- [x] - Adjusted `flow` types

## Test Plan

Example application works on `android` and `ios`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,12 +20,73 @@ or
 npm install @react-native-community/art --save
 ```
 
-### Automatic linking
+## Linking module
 
-```sh
+### Mostly automatic linking
```
Comment (thymikee) [inline README.md]: maybe mention autolinking as a preferred way?

File: docs/api.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+|  height  |   `number`   |   `0`   |
+|  width   |   `number`   |   `0`   |
+|  style   |    `any`     |   ---   |
+| children | `React.Node` |   ---   |
+
+```js
+import {Dimensions} from 'react-native';
+import {Surface} from '@react-native-community/art';
+
+const SURFACE_DIMENSIONS = Dimensions.get('window').width;
```
Comment (thymikee) [inline docs/api.md]: It's not a good practice to call dimensions outside of render, can you move it into the `Heart` function?

File: docs/api.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+```
+
+## Group
+
+Container to combine shapes or other groups into hierarchies that can be transformed as a set.
+
+|       Prop        |       Type       | Default |
+| :---------------: | :--------------: | :-----: |
+|  ...opacityProps  |  `OpacityProps`  |   ---   |
+| ...transformProps | `TransformProps` |   ---   |
```
Comment (thymikee) [inline docs/api.md]: can we link to OpacityProps and TransformProps typedefs?

File: docs/api.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+    <Surface width={SURFACE_DIMENSIONS} height={SURFACE_DIMENSIONS}>
+      <Group>{renderARTGroupContents()}</Group>
+    </Surface>
+  );
+}
+```
+
+## Shape
+
+Used to draw arbitrary vector shapes from Path. Shape implements Transform as a mixin which means it has all transform methods available for moving, scaling and rotating a shape.
```
Comment (thymikee) [inline docs/api.md]: Can we link to Path definition in svg docs?

File: docs/api.md [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+
+```js
+import {Dimensions} from 'react-native';
+import {Surface, Shape} from '@react-native-community/art';
+
+const SURFACE_DIMENSIONS = Dimensions.get('window').width;
+
+function Heart() {
+  return (
+    <Surface width={SURFACE_DIMENSIONS} height={SURFACE_DIMENSIONS}>
```
Comment (thymikee) [inline docs/api.md]: thinking about it more, I think it's gonna be more straight-forward to provide some fixed dimensions and not use Dimensions, it's an extra cognitive load.

File: docs/api.md [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+
+Generate an SVG `path` that you can pass to the `Shape` element.
+
+`constructor`
+
+```js
+new Path(path: string | Path)
+```
+
+`Path.move`
```
Comment (thymikee) [inline docs/api.md]: I'd prefer using headings, it's gonna be easier for navigation and link to certain methods
```suggestion
### `move`
```

File: docs/api.md [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+|      Prop       |      Type      | Default |
+| :-------------: | :------------: | :-----: |
+| ...opacityProps | `OpacityProps` |   ---   |
+|        x        |    `number`    |   `0`   |
+|        y        |    `number`    |   `0`   |
+|      width      |    `number`    |   `0`   |
+|     height      |    `number`    |   `0`   |
+|    children     |  `React.Node`  |   ---   |
+
+```js
```
Comment (thymikee) [inline docs/api.md]:
```suggestion
```jsx
```

File: docs/api.md (hunk 5 above)
Comment (Esemesek) [reply docs/api.md]: Good idea. Will replace it with some fixed values.

File: README.md (hunk 1 above)
Comment (Esemesek) [reply README.md]: Is there any documentation that I can link to? `0.60.0` is not released yet, so we can add it later on.

Comment (thymikee) [reply README.md]: Just running `pod install`, but yea, it's not stable yet, so let's postpone

File: docs/api.md (hunk 4 above)
Comment (Esemesek) [reply docs/api.md]: Changed example in Shape.

**Comment by thymikee** [comment]
cc @oblador, since he's using ART in [react-native-progress](https://github.com/oblador/react-native-progress) and may have some thoughts :)

**Pull request merged**